metamodel.zenta

```diff
@@ -2335,180 +2365,1111 @@ A role is assigned to human(s).
+*  If the test needs an interface for observing the actual state then create an interface for it.
+* If the behaviour is time-dependent, then dependency inject the clock
+</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="Test behaviours" id="e1557de0-91d3-4134-8e60-0ab9b1f838e4" ancestor="behaviour">
+            <documentation>* tests should test behaviour, and not module internals
+* If you take time to write the test description first, it is easier to write the test.
+* There should not be unnecessary tests (tests which are not mandated by the other rules) 
+</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="Test one thing at a time" id="55d45de7-8346-4497-98ef-f84a07973287" ancestor="behaviour">
+            <documentation>Tests should test exactly one thing
+Therefore there should be exactly one assertion in one test.</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="Pull requests" id="f6951ef3-0ef2-46c2-9859-f9ee9f84b94a" ancestor="behaviour">
+            <documentation>* pull requests should contain exactly one reference to the behaviour implemented
+* you  should include the issue number in the pull request
+* there should be different pull requests for different issues
+* it is okay to issue a pull request, fork a branch from it, continue your work, and issue a pull request in that. Just reference the second pull request from the first
+* rebase your code before issuing pull requests
+</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="framework modifications" id="3287d1a8-c265-4a2c-9efd-1754ddcfafec" ancestor="behaviour">
+            <documentation>* The CI configuration/scripts should not be changed lightly
+*  modifying the configuration and scripts determining the working of the CI should only be done with clear indication of the reasons in the pull request, and should not result in less stringent quality checks
+</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="CI is for CI" id="4c79a2cf-4a32-4290-a359-76f112f207f3" ancestor="behaviour">
+            <documentation>CI mechanisms should be used only for CI
+using the CI mechanisms for gaining unauthorized access, or for any other purpose than building and quality checking the project is strictly forbidden
+</documentation>
+          </element>
+          <element xsi:type="zenta:BasicObject" name="New dependencies" id="c0de6c9e-de2c-46d1-85f4-4e4706a8da6c" ancestor="behaviour">
+            <documentation>* New external dependencies should not be added lightly
+* While writing functionality someone else have already written is waste of effort, a dependency should be good quality and with appropriate licence.
+* Introducing a new dependency should only be done with the approval of the designer.
```

Comment:
I'm wondering if designers will be skilled enough to make this decision. It might be the responsibility of the architect.

Reply:
Let's discuss this part. What an analyst and a designer does, and how we imagine the responsibility split between them.